content/tutorials/jwt-authentication.textile

```diff
@@ -54,123 +78,156 @@ h2(#live-demo). Ably JWT auth example
     </div>
 </section>
 
-If you wish to directly check out the source code for this tutorial, we recommend you clone the tutorial repo and run it locally by following the steps below:
+If the server returns a valid JWT the client is authenticated with Ably, and you see an alert box with the message: "Client successfully connected to Ably using JWT auth".
 
-bc[sh]. git clone https://github.com/ably/tutorials.git
+The full code for this project can be found in its "Ably GitHub repository":https://github.com/ably/jwt-auth-node-tutorial.
 
-bc[sh]. git checkout jwt-authentication-nodejs
+h2(#running-locally). Running the project locally
 
-bc[sh]. cd jwt-authentication-node
+The project can be run locally using the following steps:
 
-Before running your server, make sure to replace the API Key in the 'server.js' file with your own. The DEMO will not work otherwise.
+1. Make sure that you have "Node.js":https://nodejs.org/en/ and "npm":https://www.npmjs.com/ installed on your system.
 
-bc[sh]. node server.js
+2. Clone the project:
+
+```[sh]
+git clone https://github.com/ably/jwt-auth-node-tutorial
+```
 
-Visit @localhost:3000@ in your browser
+3. Change into the project directory @jwt-auth-node-tutorial@.
+
+4. Copy @env.example@ to @.env@.
+
+5. Edit @.env@ and replace the placeholder text with your Ably API key and API secret.
```

Comment:
In the Ably dashboard, we give users a single "API Key", so I think we need to explain how they turn this into what here is being called "your Ably API key and API secret".

Looking at the tutorial code, it seems we're using the "Ably API key" as the JWT `kid` header, and the "API secret" as the JWT signing key, so I'd suggest doing the following:

- accepting a single environment variable named `API_KEY` (i.e. the Ably API key the user copies from the dashboard)
- splitting `API_KEY` on the colon character (`:`) into a `keyID` and a `keySecret`
- using `keyID` as the `kid` header in the JWT
- using `keySecret` as the second argument to `jwt.sign`

This is exactly what we do [here](https://github.com/ably-labs/ably-serverless-auth/blob/main/lambda/token/TokenSigner.js#L11-L23) in the serverless auth example, which also demonstrates using JWT tokens, but using AWS Lambda rather than an express server (see [here](https://github.com/ably-labs/ably-serverless-auth#ably-serverless-auth-example)).

Reply:
Good advice - I will make those changes.  Currently, they have to do it manually, but better in code, and as you say we only call out "API key" in the dashboard, so your approach should be less confusing and more consistent.